README.md

```diff
@@ -172,6 +172,46 @@ TID is a 32-bit integer to identify threads created with `wasi_thread_spawn`.
     For example, it can be used to indicate the main thread, which doesn't
     have a TID in the current version of this proposal.
 
+### Process
+
+* A process is a group of threads.
+
+* The main thread starts with a process which only contains
+  the main thread.
+
+* Threads created by a thread in a process using `wasi_thread_spawn`
+  is added to the process.
+
+* When a thread is terminated, it's removed from the process.
+
+### Voluntary thread termination
+
+A thread can terminate itself voluntarily by returning from
+`wasi_thread_start`.
+
+### Changes to WASI `proc_exit`
+
+With this proposal, the `proc_exit` function takes extra responsibility
+to terminate all threads in the process, not only the calling one.
+
+Any of threads in the process can call `proc_exit`.
+
+### Traps
+
+When a thread caused a trap, it terminates all threads in the process
+similarly to `proc_exit`.
+
+### Process exit status
+
+If one or more threads call WASI `proc_exit` or raise a trap,
+one of them is chosen by the runtime to represent the exit status
+of the process.
+It's non deterministic which one is chosen.
+
+If all the threads in the process have been terminated without calling
+`proc_exit` or raising a trap, it's treated as if the last thread called
+`proc_exit` with exit code 0.
```

Comment:
Should we not mention `proc_exit` here and just say that exit code is `0`? It might get a bit confusing if e.g. in the future `proc_exit` is extended with additional functionality - in that case, this sentence can be interpreted as that additional functionality should also be executed along with returning 0 as exit code (whereas I think it shouldn't).

Reply:
i can't imagine any functionalities which `proc_exit` might gain but on the other hand should not be a part of this. do you have examples?

even if we avoid mentioning `proc_exit` here, when you are adding an extra functionality to `proc_exit`, you still need to consider if the change should apply to this sentence or not.